modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/status/QuoteRequestedCommerceOrderStatusImpl.java

```diff
@@ -0,0 +1,124 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.order.status;
+
+import com.liferay.commerce.constants.CommerceOrderConstants;
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.order.CommerceOrderValidatorRegistry;
+import com.liferay.commerce.order.status.CommerceOrderStatus;
+import com.liferay.commerce.order.status.CommerceOrderStatusRegistry;
+import com.liferay.commerce.payment.method.CommercePaymentMethodRegistry;
+import com.liferay.commerce.service.CommerceOrderLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.service.WorkflowDefinitionLinkLocalService;
+
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Brian I. Kim
+ */
+@Component(
+	property = {
+		"commerce.order.status.key=" + QuoteRequestedCommerceOrderStatusImpl.KEY,
+		"commerce.order.status.priority:Integer=" + QuoteRequestedCommerceOrderStatusImpl.PRIORITY
+	},
+	service = CommerceOrderStatus.class
+)
+public class QuoteRequestedCommerceOrderStatusImpl
+	implements CommerceOrderStatus {
+
+	public static final int KEY =
+		CommerceOrderConstants.ORDER_STATUS_QUOTE_REQUESTED;
+
+	public static final int PRIORITY = 40;
+
+	@Override
+	public CommerceOrder doTransition(CommerceOrder commerceOrder, long userId)
+		throws PortalException {
+
+		commerceOrder.setOrderStatus(KEY);
+
+		commerceOrder = _commerceOrderLocalService.updateCommerceOrder(
+			commerceOrder);
+
+		return _commerceOrderLocalService.updateCommerceOrder(commerceOrder);
+	}
+
+	@Override
+	public int getKey() {
+		return KEY;
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(
+			locale, CommerceOrderConstants.getOrderStatusLabel(KEY));
+	}
+
+	@Override
+	public int getPriority() {
+		return PRIORITY;
+	}
+
+	@Override
+	public boolean isTransitionCriteriaMet(CommerceOrder commerceOrder)
+		throws PortalException {
+
+		CommerceOrderStatus currentCommerceOrderStatus =
+			_commerceOrderStatusRegistry.getCommerceOrderStatus(
+				commerceOrder.getOrderStatus());
+
+		if ((currentCommerceOrderStatus.getKey() !=
+				CommerceOrderConstants.ORDER_STATUS_CANCELLED) &&
+			(currentCommerceOrderStatus.getKey() !=
+				CommerceOrderConstants.ORDER_STATUS_COMPLETED) &&
+			(currentCommerceOrderStatus.getKey() !=
+				CommerceOrderConstants.ORDER_STATUS_DECLINED)) {
+
+			return true;
```

Comment:
@luca-pellizzon , let me know if further changes are needed. I included all statuses that I understood as "closed" order.

Reply:
I think you can use com.liferay.commerce.constants.CommerceOrderConstants#ORDER_STATUSES_OPEN